deserialize

JanetKurian1996 opened the pull request:
deserialize variable length array

JanetKurian1996 committed:
deserialize

sjames commented on serde_someip/src/de.rs:
```diff
@@ -510,10 +458,6 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
-        // Ok(value)
     }
 
     // Structs look just like maps in JSON.
```
"JSON" - seems like a leftover comment from the JSON deserializer.

sjames commented on serde_someip/src/de.rs:
```diff
@@ -601,27 +545,64 @@ impl<'de, 'a> SeqAccess<'de> for CommaSeparated<'a, 'de> {
+    }
+}
+
+struct CommaSeparatedVec<'a, 'de: 'a> {
```
naming of this type is confusing.  SOME/IP doesn't have a concept of "Comma separation".   I think the intent of this type is to iterate over a SOME/IP vector with variable number of elements.  Perhaps an appropriate name for this is SomeIpVec

JanetKurian1996 committed:
changed comma separated

JanetKurian1996 commented:
I have removed leftover comment from the JSON deserializer and changed the name of the type to iterate over a SOME/IP vector with variable number of elements.
